libtiledbsoma/src/soma/soma_dense_ndarray.cc

```diff
@@ -31,32 +31,38 @@ void SOMADenseNDArray::create(
     uint64_t index_column_size = index_column_schema->n_children;
 
     auto schema = std::make_unique<ArrowSchema>();
+    schema->name = nullptr;
     schema->format = strdup("+s");
     schema->n_children = index_column_size + 1;
     schema->dictionary = nullptr;
     schema->metadata = nullptr;
     schema->flags = 0;
     schema->release = &ArrowAdapter::release_schema;
-    schema->children = new ArrowSchema*[schema->n_children];
+    schema->children = (ArrowSchema**)malloc(
+        schema->n_children * sizeof(ArrowSchema*));
 
     std::vector<std::string> index_column_names;
     for (uint64_t dim_idx = 0; dim_idx < index_column_size; ++dim_idx) {
-        ArrowSchema* dim = schema->children[dim_idx] = new ArrowSchema;
+        ArrowSchema* dim = schema->children[dim_idx] = (ArrowSchema*)malloc(
```

Comment:
style nit: why not call `ArrowMalloc`, and `ArrowSchemaInit`?  (would allow removal of all the explicit field initialization, and prevent errors if we forget to init one of the fields.